### PR TITLE
[feature] configure lambda name & add iam outputs

### PR DESCRIPTION
**Pull request by michaelmcchen**
### Summary
I'm currently refactoring an existing piece of infrastructure to use this lambda module and having the lambda name be configurable would be very helpful so that changes can be updated via a state mv. 

Having the IAM role id be outputs also makes the policy more flexible (the consuming component can then reference the IAM role with a data resource and then add policies). 

### Test Plan
Verified that the Terraform plan (shown below) is as expected. 

```
  # module.heroku-cloudwatch-log-drain.module.lambda.aws_lambda_function.lambda will be created
  + resource "aws_lambda_function" "lambda" {
      + arn                            = (known after apply)
      + filename                       = "../../../../../shared-infra/terraform/modules/heroku-cloudwatch-log-drain/lambda_src/build/package.zip"
      + function_name                  = "heroku-log-drain-czi-connections-main"
      + handler                        = "handle"
      + id                             = (known after apply)
      + invoke_arn                     = (known after apply)
      + last_modified                  = (known after apply)
      + memory_size                    = 128
      + publish                        = false
      + qualified_arn                  = (known after apply)
      + reserved_concurrent_executions = -1
      + role                           = (known after apply)
      + runtime                        = "ruby2.5"
      + source_code_hash               = "EZlLyFz3VTf5pZdvCGR0aC49SOLOG6F4EvpDesaq+2A="
      + source_code_size               = (known after apply)
      + tags                           = {
          + "Name"      = "heroku-log-drain-czi-connections-main"
          + "env"       = "dev"
          + "managedBy" = "terraform"
          + "owner"     = "infra-eng@chanzuckerberg.com"
          + "service"   = "czi-connections-main"
        }
      + timeout                        = 15
      + version                        = (known after apply)

      + environment {
          + variables = {
              + "cloudwatch_log_group_name" = "/heroku-logs/czi-connections-main"
              + "heroku_log_drain_token"    = [redacted]
            }
        }

      + tracing_config {
          + mode = (known after apply)
        }
    }
```

### References
(Optional) Additional links to provide more context.

### CODE REVIEW

**Comment by michaelmcchen** [comment]
Hmm, the aws-aurora-postgres test seems to be broken. 
```
 Error: error creating RDS cluster: InvalidParameterCombination: Cannot find version 11.7 for aurora-postgresql
```
I believe 11.7 is supported by AWS though?

File: aws-lambda-function/main.tf [hunk 1]
```diff
@@ -1,9 +1,9 @@
 locals {
-  name = "${var.project}-${var.env}-${var.service}"
+  name = var.function_name != null ? var.function_name : "${var.project}-${var.env}-${var.service}"
 
   tags = {
     managedBy = "terraform"
-    Name      = local.name
+    Name      = var.function_name
```
Comment (ryanking) [inline aws-lambda-function/main.tf:6]: This seems the same logic as above.

File: aws-lambda-function/variables.tf [hunk 2]
```diff
@@ -72,3 +72,8 @@ variable log_retention_in_days {
   type    = number
   default = null
 }
+
+variable function_name {
+  type    = string
```
Comment (ryanking) [inline aws-lambda-function/variables.tf:77]: Add a description which documents the behavior ("if not set...").

**Review by ryanking — APPROVED** [review]

**Review by ryanking — CHANGES_REQUESTED** [review]
meant to mark "request changes"

**Review by ryanking — APPROVED** [review]

**Pull request merged**